game/scripts/vscripts/items/trumps_fists.lua

```diff
@@ -0,0 +1,175 @@
+	self.bonus_damage = self:GetAbility():GetSpecialValueFor( "bonus_damage" )
+	self.bonus_health = self:GetAbility():GetSpecialValueFor( "bonus_health" )
+	self.bonus_mana = self:GetAbility():GetSpecialValueFor( "bonus_mana" )
+
+	self.heal_prevent_duration = self:GetAbility():GetSpecialValueFor( "heal_prevent_duration" )
+
+	self.corruption_duration = self:GetAbility():GetSpecialValueFor( "corruption_duration" )
+
+	if IsServer() then
+		if self:GetParent():GetAttackCapability() == DOTA_UNIT_CAP_MELEE_ATTACK then
+			self.cold_duration = self:GetAbility():GetSpecialValueFor( "cold_duration_melee" )
+		elseif self:GetParent():GetAttackCapability() == DOTA_UNIT_CAP_RANGED_ATTACK then
+			self.cold_duration = self:GetAbility():GetSpecialValueFor( "cold_duration_ranged" )
+		else
+			self.cold_duration = 0
+		end
+	end
+end
+
+function modifier_item_trumps_fists_passive:DeclareFunctions()
+	local funcs = {
+		MODIFIER_PROPERTY_STATS_STRENGTH_BONUS,
+		MODIFIER_PROPERTY_STATS_AGILITY_BONUS,
+		MODIFIER_PROPERTY_STATS_INTELLECT_BONUS,
+		MODIFIER_PROPERTY_PREATTACK_BONUS_DAMAGE,
+		MODIFIER_PROPERTY_HEALTH_BONUS,
+		MODIFIER_PROPERTY_MANA_BONUS,
+		MODIFIER_EVENT_ON_ATTACK_LANDED,
+	}
+	return funcs
+end
+
+function modifier_item_trumps_fists_passive:GetModifierBonusStats_Strength()
+	return self.bonus_all_stats
+end
+
+function modifier_item_trumps_fists_passive:GetModifierBonusStats_Agility()
+	return self.bonus_all_stats
+end
+
+function modifier_item_trumps_fists_passive:GetModifierBonusStats_Intellect()
+	return self.bonus_all_stats
+end
+
+function modifier_item_trumps_fists_passive:GetModifierPreAttack_BonusDamage()
+	return self.bonus_damage
+end
+
+function modifier_item_trumps_fists_passive:GetModifierHealthBonus()
+	return self.bonus_health
+end
+
+function modifier_item_trumps_fists_passive:GetModifierManaBonus()
+	return self.bonus_mana
+end
+
+function modifier_item_trumps_fists_passive:OnAttackLanded( kv )
+	if IsServer() then
+		if kv.attacker == self:GetParent() then
+			kv.target:AddNewModifier( self:GetCaster(), self:GetAbility(), "modifier_item_trumps_fists_cold", { duration = self.cold_duration } )
+			kv.target:AddNewModifier( self:GetCaster(), self:GetAbility(), "modifier_item_trumps_fists_corruption", { duration = self.corruption_duration } )
+			kv.target:AddNewModifier( self:GetCaster(), self:GetAbility(), "modifier_item_trumps_fists_frostbite", { duration = self.heal_prevent_duration } )
+		end
+	end
+end
+
+--------------------------------------------------------------------------------
+
+modifier_item_trumps_fists_frostbite = class({})
+
+function modifier_item_trumps_fists_frostbite:OnCreated()
+	self.heal_prevent_percent = self:GetAbility():GetSpecialValueFor( "heal_prevent_percent" )
+end
+
+function modifier_item_trumps_fists_frostbite:IsDebuff()
+	return true
+end
+
+function modifier_item_trumps_fists_frostbite:DeclareFunctions()
+	local funcs = {
+		MODIFIER_EVENT_ON_HEAL_RECEIVED,
```

Comment:
Should this use `MODIFIER_PROPERTY_DISABLE_HEALING` and `MODIFIER_EVENT_ON_HEALTH_GAINED` (since it should prevent passive hp regen too)? No idea if health gained will trigger if healing is disabled, so you may not be able to use disable healing.

either way MODIFIER_EVENT_ON_HEALTH_GAINED is probably more accurate than MODIFIER_EVENT_ON_HEAL_RECEIVED.